pkg/collector/corechecks/snmp/report/report_utils.go

```diff
@@ -29,15 +22,40 @@ func getColumnValueFromSymbol(values *valuestore.ResultValueStore, symbol checkc
 		return nil, err
 	}
 	for index, value := range columnValues {
-		if symbol.ExtractValuePattern != nil {
-			extractedValue, err := value.ExtractStringValue(symbol.ExtractValuePattern)
-			if err != nil {
-				log.Debugf("error extracting value from `%v` with pattern `%v`: %v", value, symbol.ExtractValuePattern, err)
-				continue
-			}
-			value = extractedValue
+		newValue, err := processValue(symbol, value)
+		if err != nil {
+			continue
 		}
-		newValues[index] = value
+		newValues[index] = newValue
 	}
 	return newValues, nil
 }
+
+func processValue(symbol checkconfig.SymbolConfig, value valuestore.ResultValue) (valuestore.ResultValue, error) {
```

Comment:
Maybe the name could be more self-explanatory.

Reply:
Indeed. Update to `processValueUsingSymbolConfig` here 2f3198bd5